src/common/base/NebulaKeyUtils.h

```diff
@@ -192,6 +214,68 @@ class NebulaKeyUtils final {
         return rawKey.subpiece(0, rawKey.size() - sizeof(int64_t));
     }
 
+    static std::string encodeVariant(const VariantType& v)  {
+        switch (v.which()) {
+            case VAR_INT64:
+                return encodeInt64(boost::get<int64_t>(v));
+            case VAR_DOUBLE:
+                return encodeDouble(boost::get<double>(v));
+            case VAR_BOOL: {
+                auto val = boost::get<bool>(v);
+                std::string raw;
+                raw.reserve(sizeof(bool));
+                raw.append(reinterpret_cast<const char*>(&val), sizeof(bool));
+                return raw;
+            }
+            case VAR_STR:
+                return boost::get<std::string>(v);
+            default:
+                std::string errMsg = folly::stringPrintf("Unknown VariantType: %d", v.which());
+                LOG(ERROR) << errMsg;
+        }
+        return "";
+    }
+
+    /**
+     * Default, positive number first bit is 0, negative number is 1 .
+     * To keep the string in order, the first bit must to be inverse.
+     * for example as below :
+     *    9223372036854775807     -> "\377\377\377\377\377\377\377\377"
+     *    1                       -> "\200\000\000\000\000\000\000\001"
+     *    0                       -> "\200\000\000\000\000\000\000\000"
+     *    -1                      -> "\177\377\377\377\377\377\377\377"
+     *    -9223372036854775808    -> "\000\000\000\000\000\000\000\000"
+     */
+
+    static std::string encodeInt64(int64_t v) {
+        v ^= folly::to<int64_t>(1) << 63;
```

Comment:
How about define a constexpr?

Reply:
A good idea, Do you mean for method `encodeInt64` or constant `folly::to<int64_t>(1) << 63`?